pytroll_collectors/s3stalker.py

```diff
@@ -88,23 +111,26 @@ def _match_files_to_pattern(files, path, pattern):
     return files
 
 
-def set_last_fetch(timestamp):
-    """Set the last fetch time."""
-    DatetimeHolder.last_fetch = timestamp
-
-
-def publish_new_files(bucket, config):
+def publish_new_files(bucket, config, publisher_ready_time=2.5):
     """Publish files newly arrived in bucket."""
+    time_back = config.pop('fetch_back_to')
+    set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
     with Publish("s3_stalker") as pub:
-        time_back = config['timedelta']
-        subject = config['subject']
-        pattern = config.get('file_pattern')
-        with sleeper(2.5):
-            set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
-            s3_kwargs = config['s3_kwargs']
-            fs, files = get_last_files(bucket, pattern=pattern, **s3_kwargs)
-            messages = filelist_unzip_to_messages(fs, files, subject)
-
+        with sleeper(publisher_ready_time):
+            messages = create_messages_for_recent_files(bucket, config)
         for message in messages:
             logger.info("Publishing %s", str(message))
             pub.send(str(message))
+
+
+def create_messages_for_recent_files(bucket, config):
+    """Create messages for recent files and return."""
+    logger.debug("Create messages for recent files...")
+
+    logger.debug(f"Last fetch is {get_last_fetch()}")
+    subject = config['subject']
+    pattern = config.get('file_pattern')
+    s3_kwargs = config['s3_kwargs']
+    fs_, files = get_last_files(bucket, pattern=pattern, **s3_kwargs)
```

Comment:
I think it would be cleaner if this was outside of `create_messages_for_recent_files()`. And the `get_last_files()` could even be given just the config. The remaining call to `filelist_unzip_to_messages()` could then be renamed to `create_messages_for_recent_files()` to replace this function.

Reply:
If that's ok with you, I'd like to keep this as it is. I feel passing the config to `get_last_file` would make it's call less clear, while now we can read clearly that this function gets the last file from a bucket with the pattern and these s3 parameters.
I also this this would clutter the tests more than they already are...

Reply:
Yeah, I was thinking about that too. Having distinct arguments makes sense.

But what about moving `get_last_files()` out of the `create_new_messages_for_recent_files()` and renaming `filelist_unzip_to_messages()`?

Reply:
`filelist_unzip_to_messages` hasn't been touched in this PR and has a logical name for what it's doing IMO, that is converting a list of files (possibly zipped) to posttroll messages (see here https://github.com/pytroll/pytroll-collectors/blob/main/pytroll_collectors/fsspec_to_message.py#L77-L89). `create_new_messages_for_recent_files` on the other hand assumes that the files are recent, but that can only be the case if we call `get_last_files` first...